extensions/visual-studio-code-recent-projects/src/index.tsx

```diff
@@ -199,7 +204,29 @@ function LocalItem(props: { entry: EntryLike; uri: string; pinned?: boolean } &
 function RemoteItem(props: { entry: EntryLike; uri: string; subtitle?: string; pinned?: boolean } & PinMethods) {
   const remotePath = decodeURI(basename(props.uri));
   const scheme = getBuildScheme();
-  const uri = props.uri.replace("vscode-remote://", `${scheme}://vscode-remote/`);
+
+  const url = new URL(props.uri.replace("vscode-remote://", `${scheme}://vscode-remote/`));
+
+  let uri = url.toString();
+  let uriCloseOther = uri;
+
+  if (url.searchParams.has("windowId")) {
+    uri = url.toString();
+    url.searchParams.delete("windowId");
+    uriCloseOther = url.toString();
+  } else {
+    uriCloseOther = url.toString();
+    url.searchParams.append("windowId", "_blank");
+    uri = url.toString();
+  }
```

Comment:
Perhaps we can move this code to the `getUrl` function. The variables `uri` and `uriCloseOther` are only utilized within this specific function.